src/main/java/sage/Command/Command.java

```diff
@@ -0,0 +1,28 @@
+package sage.Command;
+
+import sage.List.TaskList;
+import sage.SageException;
+import sage.Storage;
+import sage.Ui;
+
+public abstract class Command {
+    protected boolean isExit;
+
+    public Command() {
+        this.isExit = false;
+    }
+
+    /**
+     * Executes the command with the given task list, user interface, and storage.
+     *
+     * @param tasks   The TaskList containing all tasks.
+     * @param ui      The Ui object for handling user interaction.
+     * @param storage The Storage object for saving changes to the file.
+     * @throws SageException If an error occurs during command execution.
+     */
+    public abstract void execute(TaskList tasks, Ui ui, Storage storage) throws SageException;
+
+    public boolean isExit() {
+        return isExit;
+    }
```

Comment:
Perhaps can differentiate the variable name from its getter to avoid confusion in code!